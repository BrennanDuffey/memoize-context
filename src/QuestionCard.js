import React, { Component } from 'react';
import Answer from './Answer.js';

class QuestionCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }



  render() {
    return (
      <article>
        <h3>{this.props.question}</h3>
        {this.props.answers.map((answer, index) => {
          return <Answer answer={answer} 
                         key={index} 
                         correctAnswer={this.props.correctAnswer} 
                         updateIsCorrect={this.props.updateIsCorrect} 
                         toggleShowAnswer={this.props.toggleShowAnswer}
                         updateCorrectAnswers={this.props.updateCorrectAnswers}/>
        })}
      </article>
    )
  }
}
export default QuestionCard;