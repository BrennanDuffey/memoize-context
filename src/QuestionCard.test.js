import React from 'react';
import QuestionCard from './QuestionCard.js';
import { shallow } from 'enzyme';

let mockQuestion = {
  "question": "When the context of this is described what is context actually?",
  "answers": ["The scope that this is invoked in.", "Whatever this is bound to.", "The object that is executing the javascript code."],
  "correctAnswer": "The object that is executing the javascript code.",
  "codeSnippet": null,
  "id": 1
}
let mockUpdateIsCorrect = jest.fn();
let mockToggleShowAnswer = jest.fn();
let mockUpdateCorrectAnswers = jest.fn();

describe('Answer components', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <QuestionCard  
        {...mockQuestion}
        toggleShowAnswer={mockToggleShowAnswer} 
        updateIsCorrect={mockUpdateIsCorrect} 
        updateCorrectAnswers={mockUpdateCorrectAnswers}
      />
    );
  });

  it('Should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
  });
});