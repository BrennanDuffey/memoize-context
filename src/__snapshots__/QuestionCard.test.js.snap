// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer components Should render correctly with all the information passed 1`] = `
<article
  className="question-card"
>
  <h2>
    When the context of this is described what is context actually?
  </h2>
  <Answer
    answer="The scope that this is invoked in."
    correctAnswer="The object that is executing the javascript code."
    key="0"
    toggleShowAnswer={[MockFunction]}
    updateCorrectAnswers={[MockFunction]}
    updateIsCorrect={[MockFunction]}
  />
  <Answer
    answer="Whatever this is bound to."
    correctAnswer="The object that is executing the javascript code."
    key="1"
    toggleShowAnswer={[MockFunction]}
    updateCorrectAnswers={[MockFunction]}
    updateIsCorrect={[MockFunction]}
  />
  <Answer
    answer="The object that is executing the javascript code."
    correctAnswer="The object that is executing the javascript code."
    key="2"
    toggleShowAnswer={[MockFunction]}
    updateCorrectAnswers={[MockFunction]}
    updateIsCorrect={[MockFunction]}
  />
</article>
`;
